Submitted By:            Armin K. <krejzi at email dot com>
Date:                    2013-01-10
Initial Package Version: 1.0.29
Upstream Status:         Fixed upstream
Origin:                  Upstream
Description:             Fixes building with Poppler >= 0.22.0

=== modified file 'NEWS'
--- NEWS	2013-01-02 18:23:25 +0000
+++ NEWS	2013-01-09 10:42:32 +0000
@@ -1,6 +1,11 @@
 NEWS - OpenPrinting CUPS Filters v1.0.29 - 2013-01-02
 -----------------------------------------------------

+CHANGES IN V1.0.30
+
+	- pdftoopvp: Let it build with Poppler 0.22.x. Thanks to Koji Otani
+	  from BBR Inc. (Bug #1089).
+
 CHANGES IN V1.0.29

	- Fixed ./configure option "--with-rcdir=no". Thanks to Jiri

=== modified file 'filter/pdftoopvp/OPVPOutputDev.cxx'
--- filter/pdftoopvp/OPVPOutputDev.cxx	2012-07-20 08:32:55 +0000
+++ filter/pdftoopvp/OPVPOutputDev.cxx	2013-01-09 10:42:32 +0000
@@ -1804,8 +1804,14 @@
   maskSplash = new Splash(maskBitmap, gFalse);
   maskColor[0] = 0;
   maskSplash->clear(maskColor);
+#if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
   maskSplash->drawImage(&imageSrc, &imgMaskData,
			splashModeMono8, gFalse, maskWidth, maskHeight, mat);
+#else
+  maskSplash->drawImage(&imageSrc, &imgMaskData,
+			splashModeMono8, gFalse, maskWidth, maskHeight,
+                        mat,gFalse);
+#endif
   delete imgMaskData.imgStr;
   maskStr->close();
   gfree(imgMaskData.lookup);

=== modified file 'filter/pdftoopvp/oprs/OPRS.cxx'
--- filter/pdftoopvp/oprs/OPRS.cxx	2012-09-10 19:56:59 +0000
+++ filter/pdftoopvp/oprs/OPRS.cxx	2013-01-09 10:42:32 +0000
@@ -5,6 +5,9 @@
 //========================================================================

 #include <config.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif

 #ifdef USE_GCC_PRAGMAS
 #pragma implementation
@@ -235,7 +238,11 @@
			      SplashColorMode srcMode, GBool srcAlpha,
			      int w, int h, SplashCoord *mat) {
     if (rasterMode) {
+#if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
+#else
+	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat,gFalse);
+#endif
     } else {
	return opvpSplash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
     }

